src/sinks/appsignal/mod.rs

```diff
@@ -0,0 +1,265 @@
+    const MAX_EVENTS: Option<usize> = Some(100);
+    const MAX_BYTES: Option<usize> = Some(450_000);
+    const TIMEOUT_SECS: f64 = 1.0;
+}
+
+impl_generate_config_from_default!(AppsignalSinkConfig);
+
+#[async_trait::async_trait]
+impl SinkConfig for AppsignalSinkConfig {
+    async fn build(
+        &self,
+        cx: SinkContext,
+    ) -> crate::Result<(super::VectorSink, super::Healthcheck)> {
+        let push_api_key = self.push_api_key.inner().to_string();
+        let request_settings = self.request.unwrap_with(&TowerRequestConfig::default());
+        let batch_settings = self.batch.into_batch_settings()?;
+
+        let buffer = JsonArrayBuffer::new(batch_settings.size);
+
+        let tls_settings = TlsSettings::from_options(&None)?;
+        let client = HttpClient::new(tls_settings, cx.proxy())?;
+
+        let sink = BatchedHttpSink::new(
+            self.clone(),
+            buffer,
+            request_settings,
+            batch_settings.timeout,
+            client.clone(),
+        )
+        .sink_map_err(|error| error!(message = "Fatal AppSignal sink error.", %error));
+
+        let healthcheck = healthcheck(
+            endpoint_uri(self.endpoint.clone(), "vector/healthcheck")?,
+            push_api_key,
+            client,
+        )
+        .boxed();
+
+        Ok((super::VectorSink::from_event_sink(sink), healthcheck))
+    }
+
+    fn input(&self) -> Input {
+        Input::new(DataType::Metric | DataType::Log)
+    }
+
+    fn acknowledgements(&self) -> &AcknowledgementsConfig {
+        &self.acknowledgements
+    }
+}
+
+pub struct AppsignalEventEncoder {
+    transformer: Transformer,
+}
+
+impl HttpEventEncoder<serde_json::Value> for AppsignalEventEncoder {
+    fn encode_event(&mut self, mut event: Event) -> Option<serde_json::Value> {
+        self.transformer.transform(&mut event);
+
+        match event {
+            Event::Log(log) => Some(json!({ "log": log })),
+            Event::Metric(metric) => Some(json!({ "metric": metric })),
+            _ => panic!("The AppSignal sink does not support this event: {event:?}"),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl HttpSink for AppsignalSinkConfig {
+    type Input = serde_json::Value;
+    type Output = Vec<BoxedRawValue>;
+    type Encoder = AppsignalEventEncoder;
+
+    fn build_encoder(&self) -> Self::Encoder {
+        AppsignalEventEncoder {
+            transformer: self.encoding.clone(),
+        }
+    }
+
+    async fn build_request(&self, events: Self::Output) -> crate::Result<Request<Bytes>> {
+        let uri = endpoint_uri(self.endpoint.clone(), "vector/events")?;
+        let request = Request::post(uri)
+            .header(
+                AUTHORIZATION,
+                format!("Bearer {}", self.push_api_key.inner()),
+            )
+            .header("Content-Encoding", "gzip");
+
+        let mut body = crate::serde::json::to_bytes(&events).unwrap().freeze();
```

Comment:
I see `mezmo` is doing this but I think it's preferrable to cascade up the error with `?` than to unwrap.

```suggestion
        let mut body = crate::serde::json::to_bytes(&events)?.freeze();
```